sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store.asciidoc

```diff
@@ -0,0 +1,596 @@
+* `InputT` must be convertible to value type of `OutputIteratorT`.
+* `InputT` and value type of `OutputIteratorT` must be trivially copyable
+ and default constructible.
+
+_Effects_: Stores `ElementsPerWorkItem` elements from `in` span to `out_iter`
+using the `g` group object.
+
+Properties may specify xref:data_placement[data placement].
+Default placement is a blocked one:
+`out_iter[g.get_local_linear_id() * ItemsPerWorkItem + i]` = `in[i];`
+in striped case:
+`out_iter[g.get_local_linear_id() + g.get_local_linear_range() * i]` = `in[i];`
+for `i` between `0` and `ItemsPerWorkItem`.
+Properties may also provide a hint for optimization purposes
+xref:optimization_hints[optimization_hints].
+
+=== Data Placement
+
+anchor:data_placement[]
+
+To specify a correct data placement for placing of resulting data
+there is a enum
+(proposed also in link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_sort extention`]):
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Properties:
+enum class data_placement_enum {
+  blocked,
+  striped
+};
+
+} // namespace sycl::ext::oneapi::experimental
+----
+
+Data placement layout example on group_load:
+
+* ElementsPerWorkItem = 4
+* 3 work-items in the group
+* input is: in_iter[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}
+
+Consider 2 layouts:
+
+1.`sycl::ext::oneapi::experimental::data_placement_enum::blocked`.
+
+|===
+|Work-item id|Output stored in a fixed-size array
+
+|0
+|{0, 1, 2, 3}
+|1
+|{4, 5, 6, 7}
+|2
+|{8, 9, 10, 11}
+|===
+
+2.`sycl::ext::oneapi::experimental::data_placement_enum::striped`.
+
+|===
+|Work-item id|Output stored in a fixed-size array
+
+|0
+|{0, 3, 6, 9}
+|1
+|{1, 4, 7, 10}
+|2
+|{2, 5, 8, 11}
+|===
+
+This extention adds a property that satisfies
+link:../experimental/sycl_ext_oneapi_properties.asciidoc[SYCL Properties Extension]
+requirements to identify data_placement similar to
+link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_sort extention`]
+`input_data_placement` and `output_data_placement`:
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+struct data_placement_key {
+  template <data_placement_enum Type>
+  using value_t =
+      property_value<data_placement_key,
+                     std::integral_constant<data_placement_enum, Type>>;
+};
+
+template <data_placement_enum Type>
+inline constexpr data_placement_key::value_t<Type> data_placement;
+
+inline constexpr data_placement_key::value_t<data_placement_enum::blocked> data_placement_blocked;
+inline constexpr data_placement_key::value_t<data_placement_enum::striped> data_placement_striped;
+
+} // namespace sycl::ext::oneapi::experimental
+----
+
+Specifies data layout used in group_load/store for `sycl::vec` or fixed-size
+arrays functions.
+
+Example:
+`group_load(g, input, output_span, data_placement_blocked);`
+
+=== Optimization Hints
```

Comment:
I think we should not use the word "hint" when describing these properties.  In other extensions, we use the word "hint" in cases when the property will not change the semantics of the program.  Even if you set a "hint" property wrong, the program will still run correctly.  The program may run slower when the hint is wrong, but it will never break.

We use the word "assertion" when the property could change the correctness of the program.  With an "assertion", the programmer is guaranteeing that something is true, and the program may run incorrectly if the property is set wrong.

Therefore, I'd suggest changing this section title to:

```suggestion
=== Optimization Properties
```